### PR TITLE
Breakout APNSwift into packages

### PR DESCRIPTION
**Pull request by kylebrowning**
This is the first PR among probably many to break out APNS into APNSCore, APNSAHC, APNSUrlSession, and APNS(a hyper optimized h2 client)

### CODE REVIEW

File: Sources/APNSCore/APNSClientConfiguration.swift [hunk 1]
```diff
@@ -13,17 +13,14 @@
...
 /// The configuration of an ``APNSClient``.
 public struct APNSClientConfiguration {
     /// The authentication method used by the ``APNSClient``.
     public struct AuthenticationMethod {
-        internal enum Method {
+        public enum Method {
             case jwt(privateKey: P256.Signing.PrivateKey, teamIdentifier: String, keyIdentifier: String)
-            case tls(privateKey: NIOSSLPrivateKeySource, certificateChain: [NIOSSLCertificateSource])
+            case tls
         }
```
Comment (kylebrowning) [inline Sources/APNSCore/APNSClientConfiguration.swift:24]: @FranzBusch

**Comment by vojtarylko** [comment]
@kylebrowning  Why `APNSError.reason` became `private`? The `reason` is vital for processing APNs' responses – for example sender may want to perform cleanup upon receiving `Unregistered` reason.

**Comment by kylebrowning** [comment]
@vojtarylko Im trying to remind myself why we did, but I can't think of it atm, do you wanna open a PR?

**Comment by vojtarylko** [comment]
@kylebrowning Thanks for the reply. I'll open PR.

**Pull request merged**